substrate/runtime/system/src/lib.rs

```diff
@@ -0,0 +1,378 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Substrate Demo.
+
+// Substrate Demo is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate Demo is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate Demo.  If not, see <http://www.gnu.org/licenses/>.
+
+//! System manager: Handles all of the top-level stuff; executing block/transaction, setting code
+//! and depositing logs.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg_attr(test, macro_use)] extern crate substrate_runtime_std as rstd;
+#[macro_use] extern crate substrate_runtime_support as runtime_support;
+#[cfg(test)] extern crate substrate_runtime_io as runtime_io;
+#[cfg(test)] extern crate substrate_codec as codec;
+extern crate substrate_primitives as primitives;
+
+#[cfg(feature = "std")] #[macro_use] extern crate serde_derive;
+#[cfg(feature = "std")] extern crate serde;
+
+#[cfg(any(feature = "std", test))] extern crate substrate_keyring as keyring;
+extern crate safe_mix;
+
+use rstd::prelude::*;
+use rstd::mem;
+use runtime_io::{print, storage_root, enumerated_trie_root};
+use codec::Slicable;
+use runtime_support::{Hashable, StorageValue, StorageMap};
+use primitives::{AuthorityId, Hash, BlockNumber, Header, Log};
+use primitives::block::{generic, Number as BlockNumber, Header, Log};
+use runtime::{staking, session};
+use runtime::staking::public_pass_from_payment;
+use safe_mix::TripletMix;
+
+type TxOrder = u64;
+
+storage_items! {
+	pub Nonce get(nonce): b"sys:non" => default map [ AuthorityId => TxOrder ];
+	pub BlockHashAt get(block_hash): b"sys:old" => required map [ BlockNumber => Hash ];
+	RandomSeed get(random_seed): b"sys:rnd" => required Hash;
+	// The current block number being processed. Set by `execute_block`.
+	Number get(block_number): b"sys:num" => required BlockNumber;
+	ParentHash get(parent_hash): b"sys:pha" => required Hash;
+	TransactionsRoot get(transactions_root): b"sys:txr" => required Hash;
+	Digest: b"sys:dig" => default block::Digest;
+}
+
+pub trait Checkable {
+	type CheckedType;
+	fn check(self) -> Option<Self::CheckedType>;
+}
+
+pub trait Dispatchable {
+	type AccountIdType;
+	type TxOrderType;
```

Comment:
likewise here -- associated type identifiers don't need to be suffixed with "Type"